### PR TITLE
MekHQ 1636: Fixing ForceGenerationOptionsPanel Null Pointer Issues

### PR DESCRIPTION
**Pull request by Windchild292**
This is a major refactor of ForceGenerationOptionsPanel.java to fix three different null pointer exceptions that I ran into while testing fixes for https://github.com/MegaMek/mekhq/issues/1636.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java [hunk 1]
```diff
@@ -370,15 +354,15 @@ public void updateSubfactionChoice() {
...
             for (FactionRecord fRec : RATGenerator.getInstance().getFactionList()) {
                 if (fRec.getKey().startsWith(selectedFaction.getKey() + ".")
                         && fRec.isActiveInYear(ratGenYear)) {
                     recs.add(fRec);
                 }
             }
-            Collections.sort(recs, factionSorter);
-            cbSubfaction.addItem(null); //No specific subcommand.
+            recs.sort(factionSorter);
+            cbSubfaction.addItem(null); //No specific sub-command.
```
Comment (neoancient) [inline megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java:365]: Unnecessary hyphen. Subcommand is a perfectly valid spelling.

File: megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java [hunk 2]
```diff
@@ -389,20 +373,22 @@ public void updateSubfactionChoice() {
     }
 
     /**
-     * When faction or subfaction is changed, refresh ratings combo box with appropriate
+     * When faction or sub-faction is changed, refresh ratings combo box with appropriate
```
Comment (neoancient) [inline megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java:376]: Unnecessary hyphen. Subfaction is perfectly valid.

Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java:376]: Reverted, although technically subfaction is not a word.

File: megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java:365]: Reverted, and fixed my customized autocorrect for this one.

File: megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java (hunk 2 above)
Comment (neoancient) [reply megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java]: Language is defined by use. It is an element of language that conveys meaning, and is therefore a word.

**Pull request merged**